### PR TITLE
update scan qr code page

### PR DESCRIPTION
**Pull request by jo-mut**
fixes #https://github.com/status-im/status-mobile/issues/17658

### Summary
This pr updates scan to add new contact page according to the current figma designs

### Steps to test

1. Launch app
2. Navigate to profile page
3. Tap on contacts to navigate to contacts page
4. Add new contact
5. Select scan qr code to add contact

#### Platforms

- Android
- iOS

### Figma
https://www.figma.com/file/7KIYbhoqNGAIFonE0w9TDz/Messages-for-Mobile?type=design&node-id=1816%3A381748&mode=design&t=2RahwCiMEQLWuROy-1

status: ready

### CODE REVIEW

**Comment by jo-mut** [comment]
This is a iOS screen shot showing the implementation overlaid on the figma design
Screenshot 2023-11-01 at 23 36 25

**Review by Parveshdhull — APPROVED** [review]
lgtm

File: src/status_im2/contexts/add_new_contact/views.cljs [hunk 1]
```diff
@@ -96,8 +95,9 @@
          {:type       :outline
           :icon-only? true
           :size       40
-          :on-press   #(rf/dispatch [::qr-scanner/scan-code
-                                     {:handler :contacts/qr-code-scanned}])}
+          :on-press   #(rf/dispatch [:open-modal :scan-address
+                                     {:show-subtitle?  false
+                                      :bottom-padding? false}])}
```
Comment (Parveshdhull) [inline src/status_im2/contexts/add_new_contact/views.cljs]: its little weird, we have to pass these params. Are there any instances of scanner, where we are showing subtitle and different bottom padding. Can't we just hard code them?

File: src/status_im2/contexts/wallet/scan_account/view.cljs
Comment (smohamedjavid) [inline src/status_im2/contexts/wallet/scan_account/view.cljs:1]: We shouldn't be using Wallet namespace for scanning profile QR codes.

We can create a new screen for scanning profile QR codes, which will use the common `scan-qr-code` component from the `status-im2.common.scan-qr-code.view`  namespace and does operations related to contact QR on successful read of QR.

**Review by smohamedjavid — CHANGES_REQUESTED** [review]

File: src/status_im2/contexts/add_new_contact/views.cljs (hunk 1 above)
Comment (jo-mut) [reply src/status_im2/contexts/add_new_contact/views.cljs]: @Parveshdhull creating a new space to scan profile qr code addresses concerns you highlighted in this comment

File: src/status_im2/contexts/wallet/scan_account/view.cljs
Comment (jo-mut) [reply src/status_im2/contexts/wallet/scan_account/view.cljs:1]: @smohamedjavid could you take anothe review and maybe approve

File: src/status_im2/common/scan_qr_code/view.cljs [hunk 2]
```diff
@@ -35,13 +35,14 @@
    [quo/text
     {:size   :heading-1
      :weight :semi-bold
-     :style  style/header-text}
+     :style  (style/header-text bottom-padding?)}
```
Comment (smohamedjavid) [inline src/status_im2/common/scan_qr_code/view.cljs]: Introducing a new prop `bottom-padding?` is a bit overkill. It's affecting other places too.

PR

You can use the existing `subtitle` prop to check if the subtitle exists and add the padding.

**Review by smohamedjavid — APPROVED** [review]

**Comment by jo-mut** [comment]
@status-im/mobile-qa this is ready for manual qa. Kindly have a look thanks

**Comment by pavloburykh** [comment]
> @status-im/mobile-qa this is ready for manual qa. Kindly have a look thanks

Thank you @jo-mut! Please don't forget to move PRs on the board. If PR is ready for manual QA it should be moved from Review to E2E column.

**Comment by VolodLytvynenko** [comment]
Hi @jo-mut thank you for PR. The UI of the QR code scanning page looks good. Take a look a found issue:

## ISSUE 1: [Android] Incorrect navigation after QR is scanned

### Steps:
1. Go to the 'add a contact' page
2. Scan any QR code 

Invalid QR
image

Valid QR:
image

### Actual result:

https://github.com/status-im/status-mobile/assets/52490791/6493395f-3ae7-40e9-9d22-4b4c8de247a0


### Expected result:
 The user remains on the 'Add a Contact' page after scanning a QR code
image

### Device:
- Pixel 7a, Android 13

**Comment by jo-mut** [comment]
@VolodLytvynenko I just push a fix for ISSUE: 1. You can have a second look. Thanks

https://github.com/status-im/status-mobile/assets/28704507/75c0c11c-cad9-4f69-98be-cd424520caae

**Comment by VolodLytvynenko** [comment]
Thanx. issue 1 is fixed. Unfortunatly, one more issue is introduced

## ISSUE 2: Data Not Rewritten in 'ENS or Chat Key' Textbox After QR Scan
### Steps:
1. Enter any data into the 'ENS or Chat key' textbox.
2. Scan any QR code.

### Actual result:
The entered data is not rewritten with the data contained in the QR code.

https://github.com/status-im/status-mobile/assets/52490791/8ef68239-d364-48ae-a337-0107b962a2a8

### Expected result:
The data should be rewritten to match the information contained in the scanned QR code.

https://github.com/status-im/status-mobile/assets/52490791/3928a06d-aa1e-4fa7-b1fd-76c7469148cf

**Comment by VolodLytvynenko** [comment]
## ISSUE 3: [Android] The keyboard remains opened after the scan QR page is activated
### Steps to Reproduce:
1. Focus on the 'ENS or chat key'  textbox (so the keyboard is opened).
2. Tap on the 'scan' icon.

### Actual Result:
- The keyboard remains activated and is shown on the opened 'scan QR' page.
- The upper section of the current page is displayed without opacity (this happens randomly).

https://github.com/status-im/status-mobile/assets/52490791/e8c6d23a-2048-4fc9-b835-cef6d4181a01



### Expected Result:
- The keyboard should be closed when switching to the 'scan QR' page.

### Device:
Real device: Pixel 7a, Android 13
Emul studio: Pixel 6 Pro API 31

**Comment by jo-mut** [comment]
@VolodLytvynenko I am looking into these issues. I will let you know once i resolve them. Thanks

**Comment by J-Son89** [comment]
@briansztamfater can you check this pr please, the code here seems very similar to what you added for wallet and perhaps there is a nice way for good reusability of code here. Although I think @smohamedjavid already flagged this to some degree 🤷‍♂️ 👍

**Comment by jo-mut** [comment]
@VolodLytvynenko the issues pointed out are now resolved but this pr overlaps with https://github.com/status-im/status-mobile/pull/17829 as you mentioned. I think they can be combined and merged as one

https://github.com/status-im/status-mobile/assets/28704507/f3c4d477-dfd0-45e1-a022-46b94c9645ac

**Comment by VolodLytvynenko** [comment]
hi @jo-mut the issue  https://github.com/status-im/status-mobile/pull/17767#issuecomment-1813216544 related to opened keyboard is still reproducible on Android after merge of https://github.com/status-im/status-mobile/pull/17829. Could you take a look?

**Comment by jo-mut** [comment]
Hi @VolodLytvynenko  when i get back from time off i will take a look. Thank you

**Comment by jo-mut** [comment]
@VolodLytvynenko issue 3 is also now resolved. thanks 

https://github.com/status-im/status-mobile/assets/28704507/6ee8fd88-63d6-4c16-a81c-846cb48afbbb

**Comment by VolodLytvynenko** [comment]
hi @jo-mut thank you for your work. All issues are fixed. PR is ready to be merged

**Pull request merged**